Gnome shell 48.0 vbump

listout opened the pull request:
<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

flint2 commented:
Please check https://github.com/gentoo/gentoo/pull/41116

eli-schwartz commented:
> Please check #41116

@flint2 given the co-authored-by in this PR I feel confident that the PR you refer to was already checked.

Do you have a more specific critique to make here? At the moment this isn't really actionable.

listout commented:
> Do you have a more specific critique to make here? At the moment this isn't really actionable.

The reason I made this PR was I noticed some issues on flint's PR (specifically with mutter).
On a openrc system the ebuild results in the following error during config phase:
```
...
Run-time dependency wayland-client found: YES 1.23.1
Run-time dependency wayland-cursor found: YES 1.23.1
Run-time dependency wayland-protocols found: YES 1.42
Run-time dependency wayland-egl found: YES 18.1.0

meson.build:259:4: ERROR: Problem encountered: The udev support requires logind
...
```
I fixed it by 
```diff
diff --git a/x11-wm/mutter/mutter-48.0.ebuild b/x11-wm/mutter/mutter-48.0.ebuild
index a843a9673282..ef08bf4d244a 100644
--- a/x11-wm/mutter/mutter-48.0.ebuild
+++ b/x11-wm/mutter/mutter-48.0.ebuild
@@ -174,7 +174,7 @@ src_configure() {
 		$(meson_use wayland native_backend)
 		$(meson_use screencast remote_desktop)
 		$(meson_use gnome libgnome_desktop)
-		$(meson_use udev)
+		$(meson_use udev logind)
 		-Dudev_dir=$(get_udevdir)
 		$(meson_use input_devices_wacom libwacom)
 		-Dsound_player=true
```
The libdisplay use flag was removed, but so was media-libs/libdisplay-info dependency. Unfortunately its a hard dependency now (https://gitlab.gnome.org/GNOME/mutter/-/commit/c820a63b872ddedc47dad390bc18d9c9ec5971ed)

Although -Dbash_completion=true was set, the dependency (https://gitlab.gnome.org/GNOME/mutter/-/commit/f9bb7aa2e68589d6fd5b10a8d54b211f4dc862e6) dev-python/argcomplete was not in RDEPEND nor was the completion file installed. I don't credit for finding these, devs over at #-dev helped.

mattst88 commented:
It doesn't make sense to me to make another PR for gnome-shell and mutter. Just provide feedback on flint2's PR.

listout commented:
Addressing feedback from https://github.com/gentoo/gentoo/pull/41116 @mattst88
> ```
> + -Dfonts=true
> + -Dbash_completion=true
> ```
> You've added these new configure options to the end of the array when they should be added in the same order as in meson_options.txt

I think they are in order since I see them at the end in meson.option (https://gitlab.gnome.org/GNOME/mutter/-/blob/48.1/meson.options?ref_type=tags#L222)

> Some dependencies (e.g. pango) moved in the upstream meson.build. We try to keep the same order in `*DEPEND`.

I didn't get this. I don't see much changes in meson.build between 47.4 and 48.1, I can definitely be wrong.

Norbiros commented:
@listout why you reopened this PR? Why we have 2 gnome shell 48 PRs?
